### ISSUE TITLE
Start testing out platform php templates

### ISSUE THREAD

**Issue by pirog** [lando/lando#2497]
## Conclusions
- Frameworks work with the exception of changes that need to be made at the template level
  - TYPO3 requires the redis extension is only written once to php.ini during the build #2515 
  - TYPO3 requires `trustedHostsPattern` needs to be set to `.*`  #2515 
  - Tooling such as `bin/console` in Symfony should be defined in a `.lando.base.yml` file in each template
  - EZPlatform requires branch 3.1.0; This is not currently the default branch used when selecting EZP from the UI

## Original Issue
We should start going through the php based things in the list of platforms "featured framework"
https://docs.platform.sh/frameworks/deploy-button.html
And making sure each works. For each we should

1. Fork the relevant platform.sh template for the framework
2. Get that template running on a comped platformsh site like we already have for `lando-d8` and `lando-kitchensink`
3. Try to initialize that site locally using the platform.sh integration 
4. Make any changes to the lando integration to get it to work "out of the box" if it doesnt, eg on Drupal we needed to reset the `/tmp` directory and setting `skip_permissions_hardening=1`
5. I do not think we need to add each to the functional test suite but we definitely want to be able to `lando init` each so we can easily start working on bugs that are reported.

Here are the frameworks we should be testing in order of importance:

- [x] Drupal 8
- [x] WordPress 
- [x] Magento 
- [x] Drupal 7
- [x] TYPO3
- [x] eZ Platform
- [x] Symfony

## Platform.sh Projects
- lando-d8
- lando-d7
- lando-ezplatform
- lando-magento
- lando-symfony
- lando-typo3
- lando-wordpress

**Comment by lsmith77** [comment]
probably makes sense to add Laravel to that list as well.

**Comment by pirog** [comment]
@lsmith77 let's get through the list first

**Comment by lsmith77** [comment]
I will see if I can help with Symfony and Laravel

**Comment by lsmith77** [comment]
I can also poke some ez and typo3 people

**Comment by pirog** [comment]
@lsmith77 we appreciate the offer for help but given the `alpha` status of the integration i dont think we necessarily want to invite a ton of people to try it out given its still heavily WIP. 

One of the goals of this issue is for us to determine to what extent we can start asking other people to try things out. There is a blurry line between good feedback and noise.

**Comment by lsmith77** [comment]
understood .. so better wait until beta status?

**Comment by pirog** [comment]
Our current plan is to cast a wider net and encourage more usage when there is a `beta`. Until then i'd encourage reading the docs on the recipe if you havent yet:
https://docs.lando.dev/config/platformsh.html

I think this describes where we are at currently and expectations around the integration and contributing.

**Comment by mikemilano** [comment]
@pirog Regarding Symfony, env var `APP_ENV` needs to be `dev` to run symfony tools.

New projects initialized by Platform ship with `APP_ENV` set to `prod`, but users can remove that in `.platform.app.yaml`.

The other problem however is that the `app` server sets `$_SERVER['APP_ENV'] to `prod`, and the config respects that before any environment variables.

Options:

1. Direct users how to hack `config/bootstrap.php` so it does not look at `$_SERVER['APP_ENV']`

2. Unset `$_SERVER['APP_ENV']`. This would be the best user experience. I'm not sure where we would do this.

**Comment by mikemilano** [comment]
When building ezplatform, `platformsh-php/builder.js` throws an error when `primaryRoute` is built. On the `.thru()` line, `routes[0]` doesn't exist. To get around this, I modified it to this:
```
.thru(routes => routes[0] ? routes[0].url : null)
```
Needs discussion though as I am not sure that `routes[0]` should never exist in the first place.

**Comment by pirog** [comment]
I’d make sure you are up to date with master, I merged in some stuff to
better handle primary routes yesterday

**Comment by mikemilano** [comment]
@pirog I've updated the state of the templates in the issues/notes section of the initial post.

Drupal 7 and Typo3 have similar proxy issues. Typo3 began to deserve its own issue so I generated one here: #2515

Symfony and Ezplatform will run with some additional steps. Symfony requires a meeting with Larry to discuss the ENV var, and Ezplatform requires a specific branch.

The other templates are fine.
